projects/plugins/jetpack/extensions/plugins/publicize/components/connection-toggle/index.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { FormToggle } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import ConnectionIcon from '../connection-icon';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+type ConnectionToggleProps = {
+	className: string;
+	checked: boolean;
+	id: string;
+	disabled: boolean;
+	onChange: () => void;
+	serviceName: string;
+	label: string;
+	profilePicture: string;
+};
+
+const ConnectionToggle: React.FC< ConnectionToggleProps > = props => {
+	const { className, checked, id, disabled, onChange, serviceName, label, profilePicture } = props;
+
+	const wrapperClasses = classnames( 'components-connection-toggle', {
+		'is-not-checked': ! checked,
+		'is-disabled': disabled,
+	} );
+
+	return (
+		<div className={ wrapperClasses }>
+			<ConnectionIcon
+				id={ id }
+				serviceName={ serviceName }
+				label={ label }
+				profilePicture={ profilePicture }
+			/>
+			<FormToggle id={ id } className={ className } checked={ checked } onChange={ onChange } />
+		</div>
+	);
+};
```

Comment:
I think this component should be based on the `<ToggleControl />` instead of the `<FormToggle />`.

```ts
<ToggleControl
	disabled={ disabled }
	id={ id }
	onChange={ onChange }
	checked={ checked }
	label={
		<ConnectionIcon
			id={ id }
			serviceName={ serviceName }
			label={ label }
			profilePicture={ profilePicture }
		/>
	}
/>
```

<img src="https://user-images.githubusercontent.com/77539/136092511-7db68c5b-8bf1-4915-90ea-96c23acb418a.png" width="300px" />

Something that we can tackle in follow-up tasks, too.